doc/connectivity/bluetooth/api/l2cap.rst

```diff
@@ -8,6 +8,40 @@ configuration option: :kconfig:option:`CONFIG_BT_L2CAP_DYNAMIC_CHANNEL`. These c
 support segmentation and reassembly transparently, they also support credit
 based flow control making it suitable for data streams.
 
+Channels instances are represented by the :c:struct:`bt_l2cap_chan` struct which
+contains the callbacks in the :c:struct:`bt_l2cap_chan_ops` struct to inform
+when the channel has been connected, disconnected or when the encryption has
+changed.
+In addition to that it also contains the ``recv`` callback which is called
+whenever an incoming data has been received. Data received this way can be
+marked as processed by returning 0 or using
+:c:func:`bt_l2cap_chan_recv_complete` API if processing is asynchronous.
+
+.. note::
+  The ``recv`` callback is called directly from RX Thread thus it is not
+  recommended to block for long periods of time.
+
+For sending data the :c:func:`bt_l2cap_chan_send` API can be used noting that
+it may block if no credits are available, and resuming as soon as more credits
+are available.
+
+Servers can be registered using :c:func:`bt_l2cap_server_register` API passing
+the :c:struct:`bt_l2cap_server` struct which informs what ``psm`` it should
+listen to, the required security level ``sec_level``, and the callback
+``accept`` which is called to authorize incoming connection requests and
+allocate channel instances.
+
+.. literalinclude:: ../../../../samples/bluetooth/l2cap_coc_acceptor/src/main.c
+   :language: c
+   :lines: 17-48
```

Comment:
Lets use references here to comments like e.g. usb documentation does instead of code lines as they can change:
https://github.com/zephyrproject-rtos/zephyr/blob/2d72d86ba2d561d76243879dd0b29edbf9ab7608/samples/subsys/usb/common/sample_usbd_init.c#L24-L33
https://github.com/zephyrproject-rtos/zephyr/blob/2d72d86ba2d561d76243879dd0b29edbf9ab7608/doc/connectivity/usb/device_next/usb_device.rst?plain=1#L69-L73